From ea68d36e0ba27537bd2c330413d008be946fbc51 Mon Sep 17 00:00:00 2001
From: Lukas Tribus <luky-37@hotmail.com>
Date: Thu, 4 Apr 2013 12:24:16 +0200
Subject: [PATCH 02/37] MINOR: show PCRE version and JIT status in -vv

haproxy -vv shows build informations about USE flags and lib versions.
This patch introduces informations about PCRE and the new JIT feature.
It also makes USE_PCRE_JIT=1 appear in the haproxy -vv "OPTIONS".

This is useful since with the introduction of JIT we will see libpcre
related issues.
---
 Makefile      |  1 +
 src/haproxy.c | 14 ++++++++++++++
 2 files changed, 15 insertions(+)

diff --git a/Makefile b/Makefile
index f2ac373..68e2ff8 100644
--- a/Makefile
+++ b/Makefile
@@ -562,6 +562,7 @@ endif
 # JIT PCRE
 ifneq ($(USE_PCRE_JIT),)
 OPTIONS_CFLAGS  += -DUSE_PCRE_JIT
+BUILD_OPTIONS   += $(call ignore_implicit,USE_PCRE_JIT)
 endif
 endif
 
diff --git a/src/haproxy.c b/src/haproxy.c
index dd1adcd..15c14e3 100644
--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -289,6 +289,20 @@ void display_build_opts()
 #else /* USE_OPENSSL */
 	printf("Built without OpenSSL support (USE_OPENSSL not set)\n");
 #endif
+
+#ifdef USE_PCRE
+	printf("Built with PCRE version : %s", pcre_version());
+	printf("\nPCRE library supports JIT : "
+#ifndef USE_PCRE_JIT
+	       "no (USE_PCRE_JIT not set)"
+#else
+	       "yes"
+#endif
+	       "\n");
+#else
+	printf("Built without PCRE support (using libc's regex instead)\n");
+#endif
+
 	putchar('\n');
 
 	list_pollers(stdout);
-- 
1.8.1.5

